### ISSUE TITLE
Problematic code for no-sequences showing and being flagged as non-problematic

### ISSUE THREAD

**Issue by brettz9** [eslint/eslint#5536]
**What version of ESLint are you using?**

The ESLint demo (and looking at docs in `master`).

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**

The ESLint demo (and looking at docs in `master`).

**What did you do? Please include the actual source code causing the issue.**

I noticed that the following code in `no-sequences`:

```js
(0,eval)("doSomething();");
```

...is listed in the introductory block referring to the problem of sequences as well as in a non-problematic section.

**What did you expect to happen?**

I'd expect that the code would not appear in the non-problematic block in the docs but would be flagged in the demo. I'd also expect a non-problematic equivalent to be documented.

**What actually happened? Please include the actual, raw output from ESLint.**

The code appears in a non-problematic block and was not flagged as a problem (with `no-sequences` enabled) in the demo.

**Comment by nzakas** [comment]
Sequences are allowed when enclosed in parentheses. See http://eslint.org/docs/rules/no-sequences#rule-details. The rule is behaving as expected.

The introductory paragraph and code example does not say all of the code is considered bad, it is just showing examples of code sequences. We can maybe make that clearer.

**Comment by brettz9** [comment]
Ok, sure--maybe a problematic equivalent too, since there's already a non-problematic version:

```js
0, eval("doSomething();");
```

**Comment by kaicataldo** [comment]
Very minor tweak to increase clarity of docs, including the problematic version of the code above. Thoughts?

**Comment by nzakas** [comment]
I like it.

**Comment by kaicataldo** [comment]
Kinda goofed by pushing to my fork first  and tried to delete it. Here's the actual PR: https://github.com/eslint/eslint/pull/5650 Sorry for the confusion :sweat_smile:

**Comment by brettz9** [comment]
Sweet, thanks, I think that was a very good way to resolve it...

**Comment by kaicataldo** [comment]
@brettz9 Glad to hear it - thanks for reporting the issue!
